test/bpf_tests/bpf_test.go

```diff
@@ -0,0 +1,459 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+//go:build privileged_tests
+
+//go:generate protoc --go_out=. trf.proto
+package bpftests
+
+import (
+	"bytes"
+	"encoding/binary"
+	"encoding/hex"
+	"errors"
+	"flag"
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"path"
+	"regexp"
+	"strconv"
+	"strings"
+	"syscall"
+	"testing"
+	"time"
+
+	"github.com/cilium/ebpf"
+	"github.com/cilium/ebpf/perf"
+	"github.com/golang/protobuf/proto"
+	"github.com/vishvananda/netlink/nl"
+	"google.golang.org/protobuf/encoding/protowire"
+
+	"github.com/cilium/cilium/pkg/byteorder"
+	"github.com/cilium/cilium/pkg/datapath/link"
+	"github.com/cilium/cilium/pkg/monitor"
+)
+
+var testPath = flag.String("bpf-test-path", "", "Path to the eBPF tests")
+
+func TestBPF(t *testing.T) {
+	if testPath == nil || *testPath == "" {
+		t.Fatal("-bpf-test-path is a required flag")
+	}
+
+	entries, err := os.ReadDir(*testPath)
+	if err != nil {
+		t.Fatal("os readdir: ", err)
+	}
+
+	pinPath := "/sys/fs/bpf/cilium-test"
+	defer func() {
+		os.RemoveAll(pinPath)
+	}()
+	if err = os.Mkdir(pinPath, 0755); err != nil {
+		t.Fatal(err)
+	}
```

Comment:
I'm wondering who should mount this BPFFS? At this moment, there's no logic to mount BPFFS inside `make -C test run_bpf_test`. When I tried it in my local environment with `umount`-ing BPFFS, it produced this error. I think it would be better to mount it in `Makefile` or here if it is not mounted yet. What do you think?

```
$ make -C test run_bpf_tests                                                                                                                                                                                      /home/yutaro/WorkSpace/cilium
make: Entering directory '/home/yutaro/WorkSpace/cilium/test'
make -C ../bpf/tests all
make[1]: Entering directory '/home/yutaro/WorkSpace/cilium/bpf/tests'
make[1]: Nothing to be done for 'all'.
make[1]: Leaving directory '/home/yutaro/WorkSpace/cilium/bpf/tests'
go test bpf_tests/*.go -c -o /tmp/bpf-tests
sudo /tmp/bpf-tests -bpf-test-path ../bpf/tests  -test.v
=== RUN   TestBPF
    bpf_test.go:55: mkdir /sys/fs/bpf/cilium-test: no such file or directory
--- FAIL: TestBPF (0.00s)
FAIL
make: *** [Makefile:79: run_bpf_tests] Error 1
make: Leaving directory '/home/yutaro/WorkSpace/cilium/test'
```

Reply:
@dylandreimerink Perhaps we should clear all maps' Pinning flags instead? Since this style of test (mainly driven from BPF C) doesn't/shouldn't access bpffs, tests can stay isolated without any risk of leaving leftover state, reducing flake potential.

All maps/progs can be accessed through Collections directly in Go, so I don't think we need to pin anything. bpffs interactions occur purely in userspace through Go code, which kind of falls outside of the framework's scope for now IMO.

Reply:
Yes, I set the `Pinning` setting on all maps to none, so we don't pin the maps during testing.